core/src/main/java/com/novoda/noplayer/exoplayer/ExoPlayerTwoFacade.java

```diff
@@ -0,0 +1,317 @@
+        this.mediaSourceFactory = mediaSourceFactory;
+    }
+
+    public boolean getPlayWhenReady() {
+        if (hasPlayer()) {
+            return exoPlayer.getPlayWhenReady();
+        } else {
+            return false;
+        }
+    }
+
+    public long getPlayheadPosition() {
+        if (hasPlayer()) {
+            return exoPlayer.getCurrentPosition();
+        } else {
+            return 0;
+        }
+    }
+
+    private boolean hasPlayer() {
+        return exoPlayer != null;
+    }
+
+    public int getVideoWidth() {
+        return videoWidth;
+    }
+
+    public int getVideoHeight() {
+        return videoHeight;
+    }
+
+    public long getMediaDuration() {
+        if (hasPlayer()) {
+            return exoPlayer.getDuration();
+        } else {
+            return 0;
+        }
+    }
+
+    @Override
+    public void onVideoEnabled(DecoderCounters counters) {
+        // no-op
+    }
+
+    @Override
+    public void onVideoDecoderInitialized(String decoderName, long initializedTimestampMs, long initializationDurationMs) {
+        // no-op
+    }
+
+    @Override
+    public void onVideoInputFormatChanged(Format format) {
+        // no-op
+    }
+
+    @Override
+    public void onDroppedFrames(int count, long elapsedMs) {
+        if (infoListeners == null) {
+            return;
+        }
+        infoListeners.onDroppedFrames(count, elapsedMs);
+    }
+
+    @Override
+    public void onVideoSizeChanged(int width, int height, int unappliedRotationDegrees, float pixelWidthHeightRatio) {
+        videoHeight = height;
+        videoWidth = width;
+        for (Listener listener : listeners) {
+            listener.onVideoSizeChanged(width, height, unappliedRotationDegrees, pixelWidthHeightRatio);
+        }
+    }
+
+    @Override
+    public void onRenderedFirstFrame(Surface surface) {
+        // no-op
+    }
+
+    @Override
+    public void onVideoDisabled(DecoderCounters counters) {
+        // no-op
+    }
+
+    public int getBufferedPercentage() {
+        if (hasPlayer()) {
+            return exoPlayer.getBufferedPercentage();
+        } else {
+            return 0;
+        }
+    }
+
+    public void setPlayWhenReady(boolean playWhenReady) {
+        if (hasPlayer()) {
+            exoPlayer.setPlayWhenReady(playWhenReady);
+        }
+    }
+
+    public void seekTo(long positionMillis) {
+        if (hasPlayer()) {
+            exoPlayer.seekTo(positionMillis);
+        }
+    }
+
+    public void release() {
+        if (hasPlayer()) {
+            exoPlayer.release();
+        }
+    }
+
+    public void prepare(Uri uri, ContentType contentType) {
+        exoPlayer.addListener(listener);
+
+        setPlayWhenReady(true);
+
+        MediaSource mediaSource = mediaSourceFactory.create(contentType, uri, eventListener, mediaSourceEventListener);
+        exoPlayer.prepare(mediaSource, true, false);
+    }
+
+    private final ExoPlayer.EventListener listener = new ExoPlayer.EventListener() {
```

Comment:
Also this class is pretty big, you could extract some of these implementation to external classes and have them all in a package :)